Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Config/MRTKOculusConfig.cs

```diff
@@ -0,0 +1,297 @@
+//LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+//------------------------------------------------------------------------------ -
+
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace prvncher.MixedReality.Toolkit.Config
+{
+#if UNITY_EDITOR
+    [UnityEditor.InitializeOnLoad]
+#endif
+    [CreateAssetMenu(menuName = "MRTK-Quest/MRTK-OculusConfig")]
+    public class MRTKOculusConfig : ScriptableObject
+    {
+        /// <summary>
+        /// Enum used for controlling the teleport pointer activated by MRTK Quest controllers.
+        /// </summary>
+        public enum TeleportPointerMode
+        {
+            Custom,
+            Official,
+            None
+        }
+
+        private static MRTKOculusConfig instance;
+        public static MRTKOculusConfig Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    // Creating a configurable instance so that making any runtime changes won't affect the original. 
+                    MRTKOculusConfig configFile = Resources.Load<MRTKOculusConfig>("MRTK-OculusConfig");
+                    instance = Instantiate(configFile);
+
+                    if (instance == null)
+                    {
+                        UnityEngine.Debug.LogError("Failure to detect MRTK-OculusConfig. Please create an instance using the asset context menu, and place it in any Resources folder.");
+                    }
+                }
+                return instance;
+            }
+        }
+
+        [Header("Config")]
+        [SerializeField]
+        [Tooltip("Using avatar hands requires a local avatar prefab. Failure to provide one will result in nothing being displayed. \n\n" +
+                 "Note: In order to render avatar hands, you will need to set an app id in OvrAvatarSettings. Any number will do, but it needs to be set.")]
+        private bool renderAvatarHandsInsteadOfControllers = true;
+
+        /// <summary>
+        /// Using avatar hands requires a local avatar prefab. Failure to provide one will result in nothing being displayed.
+        /// </summary>
+        public bool RenderAvatarHandsInsteadOfController => renderAvatarHandsInsteadOfControllers;
+
+
+#if OCULUSINTEGRATION_PRESENT
+        [Header("Prefab references")]
+        [SerializeField]
+        [Tooltip("Prefab reference for OVRCameraRig to load, if none are found in scene.")]
+        private OVRCameraRig ovrCameraRigPrefab = null;
+
+        /// <summary>
+        /// Prefab reference for OVRCameraRig to load, if none are found in scene.
+        /// </summary>
+        public OVRCameraRig OVRCameraRigPrefab => ovrCameraRigPrefab;
+#endif
+        [SerializeField]
+        [Tooltip("Use this if you want to manage the avatar hands prefab yourself.")]
```

Comment:
Nit: Is there a better naming for this param/doccomment? i.e. I don't understand what this means (yourself vs automatic?)